node/node.go

```diff
@@ -256,19 +255,28 @@ func (node *Node) addPendingStakingTransactions(newStakingTxs staking.StakingTra
 func (node *Node) AddPendingStakingTransaction(
 	newStakingTx *staking.StakingTransaction,
 ) error {
+	if !node.isPreStakingEpoch() {
+		utils.Logger().Info().Str("Hash", newStakingTx.Hash().Hex()).
+			Msg("Drop Staking Tx while not in Staking Epoch")
+		return nil
+	}
 	if node.NodeConfig.ShardID == shard.BeaconChainShardID {
 		errs := node.addPendingStakingTransactions(staking.StakingTransactions{newStakingTx})
 		for i := range errs {
 			if errs[i] != nil {
 				return errs[i]
 			}
 		}
-		utils.Logger().Info().Str("Hash", newStakingTx.Hash().Hex()).Msg("Broadcasting Staking Tx")
-		node.tryBroadcastStaking(newStakingTx)
 	}
+	utils.Logger().Info().Str("Hash", newStakingTx.Hash().Hex()).Msg("Broadcasting Staking Tx")
+	node.tryBroadcastStaking(newStakingTx)
 	return nil
 }
 
+func (node *Node) isPreStakingEpoch() bool {
+	return node.Blockchain().Config().IsPreStaking(node.Blockchain().CurrentHeader().Epoch())
```

Comment:
Would it be better to use Beaconchain?

Reply:
Need to use beaconchain

Reply:
Sorry don't quite get it. New logic for `AddPendingStakingTransaction`:

```
if is not pre staking:
    do nothing
elif node is on beacon chain:
    Add to local tx pool
    broadcast staking tx
elif not on beacon chain:
    broadcast staking tx to beacon chain
```

Is there any logic wrong in this pseudocode?

Reply:
epochs are only defined by the beacon chain for staking era. `node.Blockchain()` can be a shard chain and thus be wrong.